src/stdlib/parse_auditd.rs

```diff
@@ -0,0 +1,432 @@
+use std::collections::BTreeMap;
+
+use crate::{btreemap, compiler::prelude::*};
+use bytes::BytesMut;
+use chrono::DateTime;
+use linux_audit_parser::{Number, Value as AuditValue};
+
+fn parse_auditd(bytes: Value) -> Resolved {
+    let bytes = bytes.try_bytes()?;
+    // check if bytes ends with newline, otherwise append it
+    // TODO: make the parser accept bytes without newline in the linux_audit_parser crate (to-be-contributed)
+    let bytes = if bytes.last() == Some(&b'\n') {
+        bytes
+    } else {
+        let mut bytes = BytesMut::from(bytes);
+        bytes.extend_from_slice(b"\n");
+        bytes.freeze()
+    };
+    let parsed = linux_audit_parser::parse(&bytes, false)?;
+
+    let mut log = ObjectMap::new();
+
+    let timestamp_millis = i64::try_from(parsed.id.timestamp)
+        .map_err(|_| Error::TimestampOverflow(parsed.id.timestamp))?;
+    // Should we use UTC as the timezone? The logs are generated with the system's
+    // timezone... What is the correct behavior? Maybe the system where vector is running
+    // have a different timezone than the system that generated the logs... so it is
+    // not correct to assume that the current system's timezone is the correct one
+    // (with TimeZone::timestamp_millis_opt)
+    // TODO: we should document that the timestamp is parsed into UTC timezone.
+    // TODO: Maybe we should accept in a parameter a custom timezone to parse the timestamp?
```

Comment:
I think we should rely on the `timezone` that is injected into the VRL runtime. See an example in the `parse_syslog` function: https://github.com/vectordotdev/vrl/blob/c6f22a113253dbbff62abea646955ac62e819913/src/stdlib/parse_syslog.rs#L6-L19

This timezone is configurable in Vector (e.g. on the `remap` transform: https://vector.dev/docs/reference/configuration/transforms/remap/#timezone).